### PR TITLE
HDDS-7525. Migrate use of MD5 to other more secure crypto hash functions

### PR DESCRIPTION
**Pull request by GAOJHIHCYUAN**
## What changes were proposed in this pull request?

Safer alternatives, such as SHA-256, SHA-512, SHA-3 are recommended, and for password hashing, it’s even better to use algorithms that do not compute too "quickly".

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7525

## How was this patch tested?

(Please explain how this patch was tested. Ex: unit tests, manual tests)
(If this patch involves UI changes, please attach a screen-shot; otherwise, remove this)

### CODE REVIEW

**Review by GAOJHIHCYUAN — COMMENTED** [review]
Safer alternatives, such as SHA-256, SHA-512, SHA-3 are recommended, and for password hashing, it’s even better to use algorithms that do not compute too "quickly"

**Review by jojochuang — APPROVED** [review]

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ protected void execute(OzoneClient client, OzoneAddress address)
 
     if (isVerbose() && !"/dev/null".equals(dataFile.getAbsolutePath())) {
       try (InputStream stream = new FileInputStream(dataFile)) {
-        String hash = DigestUtils.md5Hex(stream);
+        String hash = DigestUtils.sha256Hex(stream);
         out().printf("Downloaded file hash : %s%n", hash);
```
Comment (kaijchen) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:94]: Please include the hash type in terminal output, for example:

```
        out().printf("Downloaded file hash (sha256): %s%n", hash);
        out().printf("Downloaded file sha256 checksum: %s%n", hash);
```

Same in PutKeyHandler.

Comment (kerneltime) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:94]: @GAOJHIHCYUAN I can merge this in once you address @kaijchen 's review comments

Comment (GAOJHIHCYUAN) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:94]: I had another PR in #4106
including the hash type in terminal output.

**Comment by kaijchen** [comment]
I'm closing this since #4106 has been merged. @GAOJHIHCYUAN next time you can add commits to PR branch instead of open a new PR.

**Pull request closed without merging**